src/relay.cpp

```diff
@@ -0,0 +1,196 @@
+// Copyright 2021 Mateusz Lichota
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#include <rclcpp/rclcpp.hpp>
+
+#include <rclcpp/generic_publisher.hpp>
+#include <rclcpp/generic_subscription.hpp>
+
+#include <memory>
+#include <string>
+
+using namespace std::chrono_literals;
+
+const auto LOOP_PERIOD = 100ms;
+
+namespace topic_tools
+{
+
+  class RelayNode : public rclcpp::Node
+  {
+  public:
+    RelayNode(
+      const std::string node_name, 
+      const rclcpp::NodeOptions &options);
+      
+    RelayNode(
+      const std::string node_name, 
+      const std::string input_topic, 
+      const std::string output_topic);
+
+  private:
+    void incoming_message_callback(std::shared_ptr<rclcpp::SerializedMessage> msg);
+    void timer_callback();
```

Comment:
Can you name this more specifically to "what it does", rather than "how it's used"? Maybe `discover_topics` or something like that - the functionality looks discovery-related.